tests/benchmarks/conftest.py

```diff
@@ -7,3 +7,21 @@ def benchmark(pytestconfig, request):
     if bench_plugin is not None:
         return bench_plugin.benchmark(request)
     return lambda request: pytest.skip("pytest-benchmark plugin not installed")
+
+
+def pytest_runtest_setup(item):
+    """Ensure that the assembly cache and lazy evaluation are off for tests."""
+    from firedrake import parameters
+    from functools import partial
+
+    def _reset(cache, lazy):
```

Comment:
Not immediately important, but at one point we may want to consider adding a *dynamic scoping* library.